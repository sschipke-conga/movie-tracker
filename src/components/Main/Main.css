/* main {
  display: flex;
  align-items: baseline;
  justify-content: space-around;
  overflow-x: scroll;
} */