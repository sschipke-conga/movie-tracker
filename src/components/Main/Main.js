// import React from 'react';
// import MovieList from '../MovieList/MovieList';
// import './Main.css';


// const Main = () => {

//   return (
//     <main>
//       <h1>Hot movies this week</h1>
//       <MovieList className = 'current-movies' />
//       <h1>Upcomming Releases</h1>
//       <MovieList className = 'upcoming-movies' />
//     </main>
//   )
// }

// export default Main;