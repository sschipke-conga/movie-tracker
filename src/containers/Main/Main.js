import React from 'react';
import MovieList from '../../components/MovieList/MovieList';
import { connect } from 'react-redux';
import './Main.css';

export const Main = ({movies, upcomingMovies, toggleFavorites}) => {
  return (
    <main>
      <h1 className = 'main__heading-playing'>Playing this week</h1>
        <MovieList toggleFavorites={toggleFavorites} className = 'current-movies' movies={movies}/>
      <h1 className = 'main__heading-upcoming'>Upcoming Releases</h1>
      <MovieList toggleFavorites={toggleFavorites} className = 'upcoming-movies' movies={upcomingMovies} />
    </main>
  )
}

export const mapStateToProps =  state => ({
  movies: state.movies,
  upcomingMovies: state.upcomingMovies
})

export default connect(mapStateToProps)(Main);