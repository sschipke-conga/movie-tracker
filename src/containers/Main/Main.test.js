import React from 'react';
import { shallow } from 'enzyme';
import {Main, mapStateToProps } from './Main';

describe ('Main',() => {
  const mockMovies = [
    {
      "poster_path": "/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg",
      "id": 290859,
      "title": "Terminator: Dark Fate",
      "vote_average": 7.4,
      "overview": "More than two decades have passed since Sarah Connor prevented Judgment Day, changed the future, and re-wrote the fate of the human race. Dani Ramos is living a simple life in Mexico City with her brother and father when a highly advanced and deadly new Terminator – a Rev-9 – travels back through time to hunt and kill her. Dani's survival depends on her joining forces with two warriors: Grace, an enhanced super-soldier from the future, and a battle-hardened Sarah Connor. As the Rev-9 ruthlessly destroys everything and everyone in its path on the hunt for Dani, the three are led to a T-800 from Sarah’s past that may be their last best hope.",
      "release_date": "2019-11-01"
    },
    {
      "poster_path": "/pIcV8XXIIvJCbtPoxF9qHMKdRr2.jpg",
      "id": 338967,
      "title": "Zombieland: Double Tap",
      "vote_average": 7.3,
      "overview": "The group will face a new zombie threat as a new breed of zombie has developed. This new super-zombie type is faster, bigger, and stronger than the previous strain of zombies and harder to kill. These super-zombies have started grouping up into a horde going from city to city leaving a path of destruction behind them.",
      "release_date": "2019-10-18"
    }
  ];
  const mockUpcomingMovies = mockMovies;

  const wrapper = shallow(<Main 
                          movies={mockMovies}
                          upcomingMovies={mockUpcomingMovies}
                          />)

  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

})

describe('Main with Redux',() => {
  const mockMovies = [
    {
      "poster_path": "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
      "id": 475557,
      "title": "Joker",
      "vote_average": 8.6,
      "overview": "During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.",
      "release_date": "2019-10-04"
    },
    {
      "poster_path": "/tBuabjEqxzoUBHfbyNbd8ulgy5j.jpg",
      "id": 420809,
      "title": "Maleficent: Mistress of Evil",
      "vote_average": 7.1,
      "overview": "Maleficent and her goddaughter Aurora begin to question the complex family ties that bind them as they are pulled in different directions by impending nuptials, unexpected allies, and dark new forces at play.",
      "release_date": "2019-10-18"
    }
  ];
  const mockUpcomingMovies = [
    {
      "poster_path": "/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg",
      "id": 290859,
      "title": "Terminator: Dark Fate",
      "vote_average": 7.4,
      "overview": "More than two decades have passed since Sarah Connor prevented Judgment Day, changed the future, and re-wrote the fate of the human race. Dani Ramos is living a simple life in Mexico City with her brother and father when a highly advanced and deadly new Terminator – a Rev-9 – travels back through time to hunt and kill her. Dani's survival depends on her joining forces with two warriors: Grace, an enhanced super-soldier from the future, and a battle-hardened Sarah Connor. As the Rev-9 ruthlessly destroys everything and everyone in its path on the hunt for Dani, the three are led to a T-800 from Sarah’s past that may be their last best hope.",
      "release_date": "2019-11-01"
    },
    {
      "poster_path": "/pIcV8XXIIvJCbtPoxF9qHMKdRr2.jpg",
      "id": 338967,
      "title": "Zombieland: Double Tap",
      "vote_average": 7.3,
      "overview": "The group will face a new zombie threat as a new breed of zombie has developed. This new super-zombie type is faster, bigger, and stronger than the previous strain of zombies and harder to kill. These super-zombies have started grouping up into a horde going from city to city leaving a path of destruction behind them.",
      "release_date": "2019-10-18"
    }
  ];

  it('mapStateToProps give all the movies in state',() => {
    const mockState = { 
      movies: mockMovies,
      upcomingMovies: mockUpcomingMovies,
      favorites:[],
      user:{id:1}
      };
    const expected = {
      movies: mockState.movies,
      upcomingMovies: mockState.upcomingMovies,
    };
    const mappedProps = mapStateToProps(mockState)
   
    expect(mappedProps).toEqual(expected)
  });




})