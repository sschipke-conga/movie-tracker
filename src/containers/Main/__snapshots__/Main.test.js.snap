// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match the snapshot with all the data passed in correctly 1`] = `
<main>
  <h1
    className="main__heading-playing"
  >
    Playing this week
  </h1>
  <MovieList
    className="current-movies"
    movies={
      Array [
        Object {
          "id": 290859,
          "overview": "More than two decades have passed since Sarah Connor prevented Judgment Day, changed the future, and re-wrote the fate of the human race. Dani Ramos is living a simple life in Mexico City with her brother and father when a highly advanced and deadly new Terminator – a Rev-9 – travels back through time to hunt and kill her. Dani's survival depends on her joining forces with two warriors: Grace, an enhanced super-soldier from the future, and a battle-hardened Sarah Connor. As the Rev-9 ruthlessly destroys everything and everyone in its path on the hunt for Dani, the three are led to a T-800 from Sarah’s past that may be their last best hope.",
          "poster_path": "/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg",
          "release_date": "2019-11-01",
          "title": "Terminator: Dark Fate",
          "vote_average": 7.4,
        },
        Object {
          "id": 338967,
          "overview": "The group will face a new zombie threat as a new breed of zombie has developed. This new super-zombie type is faster, bigger, and stronger than the previous strain of zombies and harder to kill. These super-zombies have started grouping up into a horde going from city to city leaving a path of destruction behind them.",
          "poster_path": "/pIcV8XXIIvJCbtPoxF9qHMKdRr2.jpg",
          "release_date": "2019-10-18",
          "title": "Zombieland: Double Tap",
          "vote_average": 7.3,
        },
      ]
    }
  />
  <h1
    className="main__heading-upcoming"
  >
    Upcoming Releases
  </h1>
  <MovieList
    className="upcoming-movies"
    movies={
      Array [
        Object {
          "id": 290859,
          "overview": "More than two decades have passed since Sarah Connor prevented Judgment Day, changed the future, and re-wrote the fate of the human race. Dani Ramos is living a simple life in Mexico City with her brother and father when a highly advanced and deadly new Terminator – a Rev-9 – travels back through time to hunt and kill her. Dani's survival depends on her joining forces with two warriors: Grace, an enhanced super-soldier from the future, and a battle-hardened Sarah Connor. As the Rev-9 ruthlessly destroys everything and everyone in its path on the hunt for Dani, the three are led to a T-800 from Sarah’s past that may be their last best hope.",
          "poster_path": "/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg",
          "release_date": "2019-11-01",
          "title": "Terminator: Dark Fate",
          "vote_average": 7.4,
        },
        Object {
          "id": 338967,
          "overview": "The group will face a new zombie threat as a new breed of zombie has developed. This new super-zombie type is faster, bigger, and stronger than the previous strain of zombies and harder to kill. These super-zombies have started grouping up into a horde going from city to city leaving a path of destruction behind them.",
          "poster_path": "/pIcV8XXIIvJCbtPoxF9qHMKdRr2.jpg",
          "release_date": "2019-10-18",
          "title": "Zombieland: Double Tap",
          "vote_average": 7.3,
        },
      ]
    }
  />
</main>
`;
